shell/browser/electron_download_manager_delegate.cc

```diff
@@ -130,6 +244,18 @@ void ElectronDownloadManagerDelegate::OnDownloadPathGenerated(
     const bool offscreen = !web_preferences || web_preferences->IsOffscreen();
     settings.force_detached = offscreen;
 
+#if BUILDFLAG(IS_WIN)
+    std::wstring extension = settings.default_path.FinalExtension();
+    if (!extension.empty() && settings.filters.empty()) {
+      file_dialog::Filters filter_spec = FormatFilterForExtensions(
+          {base::WideToUTF8(extension)}, {""}, true, true);
+
+      for (file_dialog::Filter filter : filter_spec) {
+        settings.filters.emplace_back(filter);
+      }
+    }
```

Comment:
- _(nit)_ we should check `settings.filters.empty()` first because if that test is `true` we can avoid creating the `extension` temporary

- We can avoid a lot of `std::vector<Filter>` temporaries by not iterating through `filter_spec` and by not emplacing-back by copy. In this situationsince `filter_spec` and `settings.filters` are the same type and we know the latter is empty, there's an easy solution; eliminate `filter_spec` temporary and directly assign `settings.filters` with the return value of `FormatFilterForExtensions()`:

```suggestion
      if (settings.filters.empty()) {
        const std::wstring extension = settings.default_path.FinalExtension();
        if (!extension.empty()) {
          settings.filters = FormatFilterForExtensions(
              {base::WideToUTF8(extension)}, {""}, true, true);
        }
      }
```